tools/autograd/templates/Functions.cpp

```diff
@@ -1,2880 +1,19 @@
-// NB: Must be at the top of file to avoid including the deprecated "math.h".
-// https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio
-#ifdef _MSC_VER
-#ifndef _USE_MATH_DEFINES
-#define _USE_MATH_DEFINES
-#endif
-#include <cmath>
-#endif
-
-#include "torch/csrc/autograd/generated/Functions.h"
-#include <ATen/Utils.h>
-#include <c10/core/TensorOptions.h>
-#include <ATen/WrapDimUtils.h>
-#include <ATen/WrapDimUtilsMulti.h>
-#include <ATen/SparseTensorUtils.h>
-#include <ATen/ExpandUtils.h>
-#include <ATen/core/Reduction.h>
-#include <ATen/Dispatch.h>
-#include <ATen/ScalarOps.h>
-
-#include <ciso646>
-#include <algorithm>
-#include <numeric>
-#include <functional>
+#include "torch/csrc/autograd/FunctionsManual.h"
 
 // ${generated_comment}
 
+// The manual function definitions that used to be here are now in torch/csrc/autograd/FunctionsManual.cpp
+// This speeds up re-compilation and allow to share these implementations so that they can be
+// used for forward formulas as well.
```

Comment:
nit: "forward formulas" -> "forward mode AD formulas", to not get confused with "forward passes"

Reply:
(feel free to address this in a follow-up, since we should try to merge this sooner rather than later to avoid merge conflicts)